src/test/ui/moves/move-fn-self-receiver.stderr

```diff
@@ -67,6 +67,10 @@ note: `Foo::use_pin_box_self` takes ownership of the receiver `self`, which move
    |
 LL |     fn use_pin_box_self(self: Pin<Box<Self>>) {}
    |                         ^^^^
+help: consider reborrowing the `Pin` instead of moving it
+   |
+LL |     pin_box_foo.as_mut().use_pin_box_self();
```

Comment:
This suggestion doesn't make sense, do we want to check that it's a `Pin<&mut _>` instead of just a `Pin`?

Reply:
Ah! I hadn't noticed that the receiver was wrong for this test. Yes, I can confirm that the receiver is indeed for `self: Pin<&mut Self>` and that filter would be enough.